### PR TITLE
Begin adding sprite tests

### PR DESCRIPTION
**Pull request by scheibo**
Confidently implementing #1369 is fairly intractable currently as there are literally thousands of possible combinations. This introduces some testing (fairly comprehensive for `getTeambuilderSpriteData`, much *much* more required for even basic levels of confidence in `getSpriteData`).

### CODE REVIEW

**Comment by scheibo** [comment]
The current game plan for #1369 is:

- this PR with initial tests
- add more thorough `getSpriteData` tests
- implement new logic for #1369, doesn't actually get used because no way to select in the UI
- add support for the selector in the UI

**Comment by Zarel** [comment]
I always thought this would be more of a burden than anything, since any refactor to this code would want to change its output, too.

**Comment by scheibo** [comment]
> I always thought this would be more of a burden than anything, since any refactor to this code would want to change its output, too.

Yes, but you need to be able to verify the output youre doing with the refactor as well.

**Comment by scheibo** [comment]
can the github workflow not write to disk or something?

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -25,6 +25,6 @@ jobs:
       with:
         node-version: ${{ matrix.node-version }}
     - run: npm install
-    - run: npm run full-test
+    - run: npm run test
```
Comment (Zarel) [inline .github/workflows/test.yml]: ?

**Comment by Zarel** [comment]
Incidentally: I would probably rather pad all dex sprites to 128x128 than have to maintain two sizes for them.

File: .github/workflows/test.yml (hunk 1 above)
Comment (scheibo) [reply .github/workflows/test.yml]: > EDIT: oh spawnSync /bin/sh ENOENT
> what fun

**Comment by scheibo** [comment]
> Incidentally: I would probably rather pad all dex sprites to 128x128 than have to maintain two sizes for them.

Out of scope for this change. :)

File: .github/workflows/test.yml (hunk 1 above)
Comment (scheibo) [reply .github/workflows/test.yml]: basically, im fine with these tests not running through CI due to not being able to download the data files. we could probably force git to do it through a submodule init or possibly by including it in the run instructions here, but its not important. having tests which you can run  at all is better than the status quo of them not existing

Comment (Zarel) [reply .github/workflows/test.yml]: Is there not an easy way to e.g. switch to `/bin/bash`?

Comment (scheibo) [reply .github/workflows/test.yml]: lmao you think bash is available but not sh?

Comment (Zarel) [reply .github/workflows/test.yml]: In hindsight yeah that's probably unlikely.

I wonder how they get away with not providing `sh`, though. I bet that breaks a lot of CI. I wouldn't be surprised if it's possible to enable somewhere in the GitHub Actions config.

**Pull request merged**